swupd/swupd.go

```diff
@@ -525,6 +526,22 @@ func UnSetHostMirror() (string, error) {
 	return unSetMirror(args, "Host")
 }
 
+// IsValidMirror checks for valid URIs that use the HTTPS or FILE protocol
+func IsValidMirror(mirror string) bool {
+	_, err := url.ParseRequestURI(mirror)
+	if err != nil {
+		return false
+	}
+
+	httpPrefix := strings.HasPrefix(strings.ToLower(mirror), "https:")
+	filePrefix := strings.HasPrefix(strings.ToLower(mirror), "file:")
+	if httpPrefix != true && filePrefix != true {
```

Comment:
@mdhorn @otaviobp So we are going to limit the user to using only **HTTPS** and **FILE** URI schemes as the swupd mirror in the Installer. Is that OK?